Modeler documentation adjustments according to docs vision

christinaausley opened the pull request:
First commit adapts Modeler platform landing page into Cloud Modeler landing page. Note that rename of overview page may require redirect (likely following merge of apis and clients into master to avoid conflicts.)

akeller commented:
Looking good so far! Will you add more commits to this? Are you looking for a review now or later?

christinaausley commented:
@akeller Still making a handful of changes here -- I was going to ask if we should make this PR a draft in that case.

If so, how do you change the PR into a draft?

akeller commented:
It's more obvious when you create it initially because you get two buttons. After that it's subtle! - https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/#:~:text=To%20convert%20a%20pull%20request,you%20to%20confirm%20the%20change.

akeller commented:
For BPMN, DMN, and Forms - we should plan to pull them out of the Camunda Modeler section and make them tool agonistic, with notes for the subtle differences in deploying these assets in Camunda Modeler (vs. Console Modeler). What do you think? Or are there significant differences I'm missing in the experience between Camunda Modeler and Console Modeler today?

nikku commented:
> For BPMN, DMN, and Forms - we should plan to pull them out of the Camunda Modeler section and make them tool agonistic, with notes for the subtle differences in deploying these assets in Camunda Modeler (vs. Console Modeler). What do you think? Or are there significant differences I'm missing in the experience between Camunda Modeler and Console Modeler today?

@akeller I think that makes loads of sense. Looking into the future it will make even more sense with basically different versions of the Modeler, one on Desktop, and one on the web.

There are some fundamental differences between both tools in some aspects thought:

* Linting / Validation and what it does
* Configuration / extending (not possible in cloud)
* Support for Camunda 7 (entirely missing in cloud)

There exist nice patterns to deal with this distinction from the users point of view, too:

* One great example I found is the [Phusion Passenger documentation](https://www.phusionpassenger.com/docs/tutorials/what_is_passenger/) which is inherently cross platform / cross-OS. It simplifies the user flow by explicitly choosing your target (Desktop vs. Console Modeler) once.

* Alternatively, what I've found useful to a certain degree, too is tabbing sections / screenshots that may look different for Desktop and Console Modeler. 

I'd personally always go for option (1): Choose my environment first and read the documentation end-to-end, no content switches necessary.

christinaausley commented:
@akeller @nikku Just merged the Forms content within Camunda Modeler into the current Forms guide I wrote so it is both Cloud and Platform friendly. I then removed the Camunda Modeler Forms content -- how do you like the notes I've made when there are mild differences between Cloud/Platform within the Forms guide?

I've also moved the BPMN and DMN components out of Camunda Modeler and adjusted element templates in the sidebar.

Let me know your thoughts.

nikku commented on docs/components/modeler/what-is-camunda-modeler.md:
```diff
@@ -1,11 +1,11 @@
 Any executable process needs a **BPMN diagram** designed and configured beforehand.
 
-The design is used to specify the flow of a process. Thereby, different events must be configured in the model so the workflow engine understands what is to be executed in this step.
+The design is used to specify the flow of a process. Thereby, different events must be configured in the model so the workflow engine understands what must executed.
```
The way I explain this: 

* You draw a BPMN diagram that represents your process, structurally / visually and is easily understood by different stakeholders.

* To implement it you add specific meta-data (_implementation details_) such as which particular implementation shall be invoked once the process reaches a task.


==> The Camunda Modeler (Desktop, Cloud, any) is, along with your IDE, the camunda implementation tool

christinaausley replied:
Adjusted our wording in What is Camunda Modeler in the latest commit -- does this help better describe Modeler's capabilities, or are there additional particular details you would add?

nikku commented on docs/components/modeler/what-is-camunda-modeler.md:
```diff
@@ -19,4 +19,6 @@ Camunda offers three tools to achieve this:
 **Cloud Modeler** and **Camunda Modeler** support the technical parameterization of the models. It's possible to design the model in Cawemo, and continue working with Cloud Modeler or Camunda Modeler.
 
-Cloud Modeler and Camunda Modeler differ mainly in their environment. Camunda Modeler is a desktop application that can be installed and used locally. Cloud Modeler is part of Cloud Console and offers a seamless integration into Camunda Cloud.
+Cloud Modeler and Camunda Modeler differ mainly in their environment. Cloud Modeler is part of Cloud Console and offers a seamless integration into Camunda Cloud to model BPMN. Camunda Modeler is a desktop application that can be installed and used locally.
```
Benefit of desktop modeler: Desktop modeler integrates in your local development environment.

christinaausley replied:
Added this detail to the latest commit -- thoughts?

nikku commented on docs/components/overview.md:
```diff
@@ -10,7 +10,7 @@ This section contains product manual content for each component in Camunda Cloud
 - [Clients](../apis-clients/overview.md) - Supported clients for working with Camunda Cloud, including information on building your own.
 - [Cloud Console](cloud-console/introduction.md) - More information on working with Cloud Console.
-- [Modeler](modeler/overview.md) - Documentation on Camunda's modeling tools, including Cloud Modeler and Camunda Modeler.
+- [Modeler](modeler/what-is-camunda-modeler.md) - Documentation on Camunda's modeling tools, including Cloud Modeler and Camunda Modeler.
```
Consider renamining to `about`?

christinaausley replied:
Do you prefer `about`? How do you feel about `using-camunda-modeler` or `working-with-camunda-modeler` or `applying-camunda-modeler`?

nikku replied:
Given that it is the overview page, I think something generic like `about` is best. Also considering that you'll not be able to go around our `modeler(s)` anyway, as you do anything Camunda.

nikku commented on docs/guides/utilizing-forms.md:
```diff
@@ -0,0 +1,71 @@
+Within Forms, we have the option to add text fields, numerical values, checkboxes, radio elements, selection menus, text components, and buttons.
+
+:::note
+Within Camunda Platform, you can also analyze [embedded forms](https://docs.camunda.org/manual/latest/reference/forms/embedded-forms/).
```
What do you refer to with _analyze_?

nikku replied:
Is platform also in the scope of this documentation?

christinaausley replied:
In latest commit, changed **analyze** to **utilize** -- thoughts?

The goal of this particular guide is to be rather tool-agnostic. I've worked to include these larger differences between Cloud/Platform in notes such as these. Do you feel there are particular areas where we should add/remove information to ensure we cover both Platform/Cloud without leaving out important information?

nikku replied:
_utilize_ is the word to use, thanks. Analyze simply does not fit.

nikku commented on docs/guides/utilizing-forms.md:
```diff
@@ -0,0 +1,71 @@
+
+## Build your form
+
+Now you can start to build your Camunda form. First, let's name our form by clicking the three vertical dots next to the text reading **New Form** in the top left corner of the page. Select **Rename** to give your form a new name. In this example, we'll analyze a form to help with a task in obtaining an email message.
```
What do you refer to with _analyze_ here?

christinaausley replied:
Adjusted to **build** here, in latest commit.

nikku commented on docs/guides/utilizing-forms.md:
```diff
@@ -0,0 +1,71 @@
+
+## Connect your form to a BPMN diagram
+
+Click on the **Forms** tab again to view a list of all the forms you've created.
```
Differs entirely in desktop modeler, as it does not help with selecting a form.

christinaausley replied:
Pulled the process for Platform out in a note admonition in the latest commit -- is there additional crucial information for Platform you would add here?

nikku reviewed (commented):
As I'm reviewing the cloud docs for the first time I'll start with some unrelated, general feedback (UI/UX wise):

* Sign up secondary action is barely visible; needs clear distinction from background:

    ![image](https://user-images.githubusercontent.com/58601/143254719-fe3e237c-6790-4b7f-9178-9e88ec83b4d8.png)

* Website footprint is huge and must be massively decreased (currently visiting the front-page downloads 18 MB in development mode; I hope that this is different in prod (did not verify it!)

    ![image](https://user-images.githubusercontent.com/58601/143255074-8019520d-72fd-41a3-a30c-338e6f1e06c6.png)

    Slim websites are a major driver for user satisfaction and conversion (to whatever goal, really).
    
----

Feedback regarding the Modeler section(s) specifically:

* Consider unifying the wording: At which point do we start to talk about "camunda Modeler", where do we stop talking about "Modeler" in general?
    
    ![image](https://user-images.githubusercontent.com/58601/143255813-a7647956-0cb6-46af-beae-8c4f0f7b6ea1.png)

* Consider _the_ vs. _no the_ to address the tool. For me _the modeler (a specific one) is a tool that we talk about like in _the TOOL offers you_ rather than _TOOL offers you_

    ![image](https://user-images.githubusercontent.com/58601/143256149-ee68b6d7-f6d0-4941-aeac-9888174d319e.png)

* Consider active vs. passive voice: _A process needs to be configured_ vs. _As a user I configure a process_. :arrow_right: I'm sure we encourage active over passive voice.

* Overall structure: Consider [unifying sections](https://github.com/camunda-cloud/camunda-cloud-documentation/pull/415#issuecomment-975209891) at some point; users want do do exactly the same thing (with the small distinction of "Launching Cloud Modeler" vs. "Installing + Launching Desktop Modeler". What they get is different capabilities, hopefully more aligned looking into the mid-term future. 

----

Specific feedback regarding the forms guide:

* :+1: for picking up the user and explaining the small (but soon to evolve) feature set. Very important

* What is clearly visible to me here is how a clear distinction between the modeler tools would help me as a user. I'm likely using one of the tools, not both at the same time.

christinaausley commented:
Thank you for your thorough review of Modeler content @nikku!

While this PR aims to largely incorporate just the components of the [docs vision](https://docs.google.com/document/d/1e9jnGJLvbM9121VQhjGfTNtONLDggalggYHIuUUzR0A/edit#heading=h.dg3n7dqfer96), I do find your comments surrounding the larger complications of Modeler valid, particularly the discrepancies between Camunda Modeler and Cloud Modeler.

@akeller how do you feel about a tabbed approach for the currently separated sections of Camunda and Cloud Modeler, similar to how we have separated experiences based on operating system in other documentation? While this may be beyond the scope of this PR, it's something we could consider for future adjustments to Modeler.

Regarding "the Camunda Modeler" versus "Camunda Modeler," we do prefer to refer to Modeler similar to other products/company brands with standalone names (which, in a way, suggest its independent tooling) -- for example, "Using Camunda Cloud" or "Using Optimize" versus "Using the Camunda Cloud" or "Using the Optimize". Does this make sense?

Please do let me know where you've noted passive voice as I'm happy to review docs that don't seem to flow according to this standard. As you mentioned, active voice does indeed follow with our style guide standards.

Nevertheless, I've made a handful of documentation fixes in the latest commit. Let me know if these resolve your outstanding comments or if there are additional, larger fixes to be made.

akeller commented:
> @akeller how do you feel about a tabbed approach for the currently separated sections of Camunda and Cloud Modeler, similar to how we have separated experiences based on operating system in other documentation? While this may be beyond the scope of this PR, it's something we could consider for future adjustments to Modeler.

I'm largely ok with this, but definitely agree it's out of scope of this PR. My goal for this quarter is to get the documentation moved over, Cloud-first & SaaS-first, then make adjustments like this. 

@nikku regarding your other comments around UX and loading, you are 100% right. I'm hiring a fullstack engineering specifically for items like this and working with the marketing & brand team to get better graphics and colors in place. Do you know of a good fullstack engineer who wants to work on Developer Experience projects? 😄

nikku commented:
> Do you know of a good fullstack engineer who wants to work on Developer Experience projects? 

I don't, but I think getting the basics right is a time-boxed activity worth executing before Camunda 8. Consider pulling in someone to help. We have a lot of knowledge in the org.

nikku commented on docs/components/modeler/what-is-camunda-modeler.md:
```diff
@@ -5,11 +5,11 @@ title: What is Camunda Modeler?
+In tandem, different events and implementation details (such as the conditions within a gateway, or the specifications of a service task) must be configured in the model so the workflow engine understands what must executed once the process reaches a certain task.
 
-Camunda offers three tools to achieve this:
+Camunda offers three tools to implement these details:
```
Core to all drawing tools is that they do both:

* draw (or design) your diagrams
* implement them (adding implementation details)

Let us phrase it this way. The tools are not meant to be "implementation details adders".

christinaausley replied:
Updated in latest commit. Thoughts?

nikku commented on docs/guides/utilizing-forms.md:
```diff
@@ -52,17 +52,21 @@ To save your form in Camunda Platform, click **File > Save File As...** in the t
 Click on the **Forms** tab again to view a list of all the forms you've created.
 
-Next, let's implement a task form into a diagram. To connect your form to a user task or start event, take the following steps:
+Take the following steps:
```
What I find interesting here is that you assume cloud to be the default (and branch of where Platform differs). I think that could work, if users know about it (upfront). But then again, why not clearly differentiate at the start (cf. my previous examples).

christinaausley replied:
Brought the note to the beginning of the section and clarified discrepancies between Camunda Modeler and Cloud Modeler in latest commit. Thoughts?

akeller replied:
I like what you did here @christinaausley. It's an awkward line to walk, but it fits in this context.

nikku reviewed (commented):
Thanks for incorporating some of my review hints. :clap:

christinaausley commented:
Thanks again for your comments @nikku. I've made several changes in the past few commits, so let me know if you feel your original comments have been resolved or if there are additional changes to make before merging.

Pulling in @akeller for final review/thoughts/changes as well.

akeller commented on docs/components/modeler/bpmn/bpmn.md:
```diff
@@ -0,0 +1,46 @@
+
+## Create new BPMN 2.0 Diagram
+
+![New BPMN Diagram](./img/quickstart-1.png)
+
+To start modeling, create a new BPMN 2.0 diagram for the Camunda Platform by selecting **Create diagram > Create new BPMN diagram (Camunda Platform)** in the top-level menu.
```
This is not Cloud-first. Can you adjust the screenshot and the opening lines here? 

I would even advocate to remove the screenshot (or have it highlight Cloud) and modify the section to be tool agnostic.

```suggestion
![New BPMN Diagram](./img/quickstart-1.png)

To start modeling, create a new BPMN 2.0 diagram by selecting **Create diagram > Create new BPMN diagram** for your process engine in the top-level menu.
```

christinaausley replied:
Updated in latest commit, but note I am drafting the "Automating a process using BPMN" doc this week, which should replace much of this content.

akeller commented on docs/components/modeler/bpmn/bpmn.md:
```diff
@@ -0,0 +1,46 @@
+
+## BPMN 2.0 Coverage
+
+The Camunda Modeler covers all BPMN 2.0 elements for modeling processes and collaborations.
```
Can you add a link to the BPMN Coverage for Cloud? 

/docs/reference/bpmn-processes/bpmn-coverage/

akeller commented on docs/components/modeler/dmn/dmn.md:
```diff
@@ -0,0 +1,61 @@
+
+## Create new DMN Decision Requirement Diagram
+
+![New DMN Diagram](./img/create-dmn.png)
+
+To start modeling, create a new DMN 1.3 diagram for the Camunda Platform by selecting **Create diagram > Create new DMN diagram (Camunda Platform)** in the top-level menu.
+
```
This is not Cloud first, similar to BPMN. Can you modify the diagram and sentence to be process engine agnostic?

christinaausley replied:
Should this be labeled Platform-only? Should we have DMN content within this PR per the docs vision, or should this component be removed for now?

akeller replied:
Since we are moving all Modeler content over, please keep it and label it Platform-only. My mistake, my brain was moving too fast and forgot we don't have DMN for Cloud quite yet!

akeller commented on docs/components/modeler/dmn/dmn.md:
```diff
@@ -0,0 +1,61 @@
+title: Editing DMN in Camunda Modeler
+---
+
+# Quickstart
```
Does quickstart here make sense? I would delete this.

akeller commented on docs/components/modeler/bpmn/bpmn.md:
```diff
@@ -0,0 +1,46 @@
+title: BPMN in Camunda Modeler
+---
+
+# Quickstart
```
Does quickstart here make sense? I would delete this.

akeller commented on docs/components/modeler/what-is-camunda-modeler.md:
```diff
@@ -0,0 +1,24 @@
+
+Camunda offers three tools to design your diagrams and implement them:
+
+- [Cawemo](https://cawemo.com/)
+- [Cloud Modeler](./cloud-modeler/launch-cloud-modeler.md)
+- [Camunda Modeler](./camunda-modeler/install-the-modeler.md)
```
I would recommend reordering this
- Cloud Modeler
- Camunda Modeler
- Cawemo

akeller commented on docs/components/modeler/what-is-camunda-modeler.md:
```diff
@@ -0,0 +1,24 @@
+- [Cloud Modeler](./cloud-modeler/launch-cloud-modeler.md)
+- [Camunda Modeler](./camunda-modeler/install-the-modeler.md)
+
+**Cawemo** focuses on the design phase and provides options for collaborative work on a BPMN model.
+
+**Cloud Modeler** and **Camunda Modeler** support the technical parameterization of the models. It's possible to design the model in Cawemo, and continue working with Cloud Modeler or Camunda Modeler.
```
And then reordering this to match.

nikku replied:
Please rename `parameterization` with `implementation` (adding implementation related meta-data, making the process executable, ...).

akeller reviewed (commented):
Left some notes here.

Have you also look at https://docs.camunda.io/docs/components/tasklist/userguide/user-interface/camunda-forms/? I'm thinking some of the information in that section makes this redundant. Can you confirm?

christinaausley commented:
@akeller @nikku Just posting an update on what has/has yet to be accomplished within this PR:
Completed:
- Forms guide
- Content migration from Platform
In progress:
- DMN update/refresh (completed EOD 12/7)
- BPMN draft/refresh (completed EOD 12/7)
- Image/attachment updates (completed EOD 12/8)

akeller commented:
@christinaausley sync with @MaxTru on the images/screenshots. He was just messaging me about where to update new screenshots today and whether they should target the Platform or Cloud repo.

MaxTru commented:
> @christinaausley sync with @MaxTru on the images/screenshots. He was just messaging me about where to update new screenshots today and whether they should target the Platform or Cloud repo.

Basic approach (please correct me if wrong :-) ):
* As soon as the Modeler documentation was migrated to .io domain, we will create a PR so that the screenshot automation scripts will create PRs in the new and not in the old repo anymore. So the old repo will outdate, but that is okay since my understanding is there will be a redirect anyhow.
* Screenshot automation is kept here: https://github.com/camunda/camunda-docs-modeler-screenshots

akeller commented on docs/components/modeler/about.md:
```diff
@@ -0,0 +1,26 @@
+
+**Cawemo** focuses on the design phase and provides options for collaborative work on a BPMN model.
+
+In this guide, we'll demonstrate modeling BPMN diagrams using both Cloud Modeler and Camunda Modeler.
```
I recommend removing this line, doesn't make sense anymore. You could consider an admonition that links to Cawemo docs -https://docs.camunda.org/cawemo/latest/.

christinaausley replied:
Agreed. Adjusted in latest commit.

akeller commented on docs/components/modeler/about.md:
```diff
@@ -0,0 +1,26 @@
+
+In this guide, we'll demonstrate modeling BPMN diagrams using both Cloud Modeler and Camunda Modeler.
+
+Additionally, the Camunda Forms feature allows you to easily design and configure forms. Once configured, they can be connected to a user task or start event to implement a task form in your application. Learn more in our documentation on [utilizing forms](./../../guides/utilizing-forms.md).
```
I like that we introduce the forms guide here! Thank you!

Do you have any idea for introducing the BPMN and DMN sections? Since the goal for these is to be tool agnostic, the content in them should be for users of Cloud Modeler & Camunda Modeler, when used with either Camunda Cloud or Camunda Platform. Maybe include a **Next steps** section? Or adjust the intro sentences to include references to BPMN and DMN?

christinaausley replied:
I think a next steps section is clean/appropriate. I'll add this for now, and we can pop in the BPMN guide once merged to master.

akeller commented on docs/components/modeler/bpmn/bpmn.md:
```diff
@@ -0,0 +1,49 @@
+
+To start modeling, create a new BPMN 2.0 diagram by selecting **Create New Diagram** in the **Diagrams** tab for your process engine in the top-level menu.
+
+If working in Platform, select **Create diagram > Create new BPMN diagram**.
```
Remove this line.

akeller replied:
Errrr.... maybe move this into an admonition and fully qualify Platform -> Camunda Platform.

akeller commented on docs/components/modeler/bpmn/bpmn.md:
```diff
@@ -0,0 +1,49 @@
+
+## Save a diagram
+
+To save your state of work in Cloud, click **Save**.
```
```suggestion
To save your state of work, click **Save**.
```

akeller commented on docs/components/modeler/bpmn/bpmn.md:
```diff
@@ -0,0 +1,49 @@
+
+To save your state of work in Cloud, click **Save**.
+
+To save your state of work in Platform, click **File > Save File As...** in the top-level menu. Then select a location on your file system to store the diagram in the BPMN 2.0 XML format. You can load that file again by clicking **File > Open File...**.
```
Move this into an admonition.

akeller commented on docs/components/modeler/dmn/dmn.md:
```diff
@@ -0,0 +1,62 @@
+description: To start modeling, let's create a new DMN diagram.
+---
+
+<span class="badge badge--CCSM">CCSM</span>
```
This is actually only available in Platform, so could you make a badge for "Camunda Platform Only". 

Given how small and not obvious the badging is for this, can you also add an admonition? Something like "DMN is only available in Camunda Platform at this time."

akeller commented on docs/components/modeler/dmn/dmn.md:
```diff
@@ -0,0 +1,62 @@
+---
+id: camunda-modeler-dmn
+title: Editing DMN in Camunda Modeler
```
Do you think we should add (Coming soon!) or something in the title? I'm thinking of someone looking at the sidebar and trying to help them understand as we add content in there, what that experience may look like?

akeller replied:
Please add a Platform-only label to this instead of Coming Soon.

akeller commented on sidebars.js:
```diff
@@ -17,6 +17,7 @@ module.exports = {
     "guides/getting-started-orchestrate-microservices",
     "guides/setting-up-development-project",
     "guides/message-correlation",
+    "guides/utilizing-forms",
```
Bump this up so its between the Microservice orchestration guide and setting up your first development project.

akeller reviewed (commented):
I'm quite happy with this, despite the 10 comments I made across all the docs! 

Don't forget to add the redirect links so we can coordinate with IT.

Great job here! So close to getting this done. Thank you for your flexibility and determination here.

akeller commented:
Adding a few other files to cover other Modeler content, you'll see a few commits from me here shortly 🙂

akeller commented:
I totally forgot I wanted to move over some of the BPMN content in Reference into the Modeler section. 

@christinaausley if you have a chance tomorrow can you move BPMN processes (the whole directory) into the BPMN section? And add the corresponding redirect? You should be able to do one for the whole directory/folder. If you don't have time, I'll take care of it. Also reach out if you can't get the redirect.

christinaausley commented:
> I totally forgot I wanted to move over some of the BPMN content in Reference into the Modeler section.
> 
> @christinaausley if you have a chance tomorrow can you move BPMN processes (the whole directory) into the BPMN section? And add the corresponding redirect? You should be able to do one for the whole directory/folder. If you don't have time, I'll take care of it. Also reach out if you can't get the redirect.

Will do today. All of the BPMN 2.0 reference material, correct?

akeller commented:
@christinaausley yes, everything in docs/reference/bpmn-processes/
